pkg/urbit/jets/d/by_apt.c

```diff
@@ -0,0 +1,88 @@
+/* j/4/by_apt.c
+**
+*/
+#include "all.h"
+
+static
+c3_y _by_apt(u3_noun a, u3_noun l, u3_noun r)
+{
+  if (a == 0) {
+    return c3y;
+  }
+
+  u3_noun n_a, l_a, r_a;
+  u3_noun p_n_a, q_n_a;
+  u3x_trel(a, &n_a, &l_a, &r_a);
+  u3x_cell(n_a, &p_n_a, &q_n_a);
+
+  u3_noun u_l;
+  if (c3y == u3r_cell(l, 0, &u_l)) {
```

Comment:
I think these `u3r_cell()` calls (and subsequent `u3r_trell()` calls) are a subtle mismatch with the hoon. `?~` is not `?@`, but really just `?:(=(0 ...) ...)`.

It seems that each of these conditionals should be shaped like so:

```
if ( u3_nul != x ) {
  u3x_cell();
  ...
} 
```